### DISCUSSION TITLE
Recovery

### DISCUSSION THREAD

**Discussion by klaw13** [idaholab/moose#21761] [Q&A Getting Started] [answered]
Hi, I'm facing issues with using the "--recover" option on my simulation.
`
*** ERROR ***
ERROR: Neither one of the following files can be located:
        '_mesh.cpr/1/header.cpr' nor
        '_mesh.cpr'
If you are running a parallel job, double check that you've created a split for 1 ranks.
Note: One of paths above may refer to a valid directory on your system, however we are attempting to read a valid header file.
`
The checkpoint file is being created in the directory I'm running the simulation from.

**Comment by GiudGiud** [comment]
Hello

do you have an execute_on flag set in the Outputs block by any chance?

Guillaume

**Reply by klaw13** [reply]
No, I do not. My output block is as follows: 

```
[Outputs]
  exodus = true
  perf_graph = true
  print_linear_residuals = true
  
  [./checkpoint]
    type = Checkpoint
    num_files = 2
    use_displaced = True
  []
  [console]
    type = Console
    all_variable_norms = true
  []
[]
```

**Reply by GiudGiud** [reply]
ok thanks different issue then.

Can you specify the prefix for the recover file? on the command line
Are you creating the checkpoint in parallel or in serial? Are you loading it with the same number of processes?

btw i think you want to use triple quotes for formatting

**Reply by klaw13** [reply]
Ah, thanks!

The checkpoint directory is hydSq_629_checkpoint_cp with a subdirectory of 5250_mesh.cpr which has a meta_data_mesh.rd file and a 1/header.cpr and 1/split-1-0.cpr 

The command I use to call the recovery is 
```
 ../../eel-opt -i hydSq_629.i --recover
```

**Reply by GiudGiud** [reply]
yes after that command you will need to pass the name of the folder. 
If you type `../../eel-opt --help` it ll say that
```
  --recover [file_base]                             Continue the calculation.  If file_base is omitted then the most recent recovery file will be utilized
```

**Comment by ykvishal** [comment, marked as answer]
I faced similar issue. When I edited folder name by replacing `checkpoint `to `out` (i.e. rename hydSq_629_checkpoint_cp to hydSq_629_out_cp), the ` --recover` option was working. 

In future, you can use this block
```
 [./out]
     type = Checkpoint
     num_files = 2
     use_displaced = True
 []
```

**Reply by klaw13** [reply]
That worked, thank you so much!

**Reply by GiudGiud** [reply]
for future reference, the folder name should be passed as an argument to --recover. `out` works here because it s the default

**Reply by klaw13** [reply]
Perfect, thank you!
